commands/command_lock.go

```diff
@@ -30,8 +30,14 @@ func lockCommand(cmd *cobra.Command, args []string) {
 
 	success := true
 	locks := make([]locking.Lock, 0, len(args))
+	lockData, err := computeLockData()
+	if err != nil {
+		lockClient.Close()
```

Comment:
I think we could avoid the need to call `lockClient.Close()` here by simply making this `computeLockData()` call further up, before the `newLockClient()` [call](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/commands/command_lock.go#L27).  That's what is already being done in this PR in the `commands/command_locks.go` source file, so if we do that here and in `commands/command_unlock.go` I think we can dodge the problem of deferred functions not running during `os.Exit()`.